bika/health/utils.py

```diff
@@ -105,3 +110,46 @@ def set_field_value(instance, field_name, value):
     if not field:
         api.fail("No field {} found for {}".format(field_name, repr(instance)))
     field.set(instance, value)
+
+
+def handle_after_submit(context, request, state):
+    """Handles actions provided in extra_buttons slot from edit forms
+    """
+    if request.get("form.button.new_sample"):
+        # Redirect to Sample Add from Patient
+        next_url = api.get_url(context)
+        if IPatient.providedBy(context):
+            uid = context.UID()
+            client = context.getPrimaryReferrer()
+            folder = client or api.get_portal().analysisrequests
+            folder_url = api.get_url(folder)
+            next_url = "{}/ar_add?Patient={}&ar_count=1".format(folder_url, uid)
+
+        # Redirect to Sample Add form from Batch
+        elif IBatch.providedBy(context):
+            next_url = "{}/ar_add?ar_count=1".format(next_url)
```

Comment:
Should not the ar_count depend on the number defined in the system by default?

Reply:
Fixed, please retest